lib/cc/engine/analyzers/helpers/main.rb

```diff
@@ -35,18 +46,42 @@ def find_other_locations(all_locations, current)
           all_locations.reject { |location| location == current }
         end
 
-        def format_location(location)
+        def format_location(issue, location)
+          current_sexp = flay.hashes[issue.structural_hash].detect do |sexp|
+            sexp.line == location.line
+          end
+
+          format_sexp(current_sexp)
+        end
+
+        def format_locations(issue, location)
```

Comment:
Could we do something where we're doing the split between `location` and `other_locations` in a single place and then have another place loop through and format? This method and the previous one looks too similar...

Here's something crazy:

```ruby
def new_violation(issue, location)
  first_sexp, other_sexps = formatted_sexps_by(location)

  {
    #...,
    "location": first_sexp,
    "other_locations": other_sexps,
  }
end

def formatted_sexps_by(location)
  flay.hashes[issue.structural_hash].partition do |sexp|
    sexp.line == location.line
  end.flatten.map do |sexp|
    format_sexp(sexp)
  end
end
```

This is technically shorter, but probably sacrifices readability...

Anyway, not a blocker.